backend/vkt/src/main/java/fi/oph/vkt/service/PublicEnrollmentService.java

```diff
@@ -122,12 +170,15 @@ public PublicEnrollmentInitialisationDTO initialiseEnrollmentToQueue(final long
       throw new APIException(APIExceptionType.INITIALISE_ENROLLMENT_DUPLICATE_PERSON);
```

Comment:
Tähän `isPersonEnrolled` logiikkaan tarvitaan varmaan jotain muutoksia. Seuraava flow aiheuttaa mielestäni virheellisesti tämän duplicate person -virheen:
1. Ilmoittaudu kokeeseen X
2. Tunnistaudu henkilönä Y
3. Täytä ilmoittautumistiedot
4. Siirry maksamaan
5. Palauta hylätty maksu paytrailissä
6. Yritä ilmoittautua uudestaan (koe X, henkilö Y)
7. Virheviesti

Tällöin siis saadaan duplicate person -virhe, vaikka ilmoittautumista ei ole oikeasti viety hyväksytysti loppuun.

Reply:
Tälle virhetapaukselle pitäisi myös jokin kunnollinen virhesivu saada. Tällä hetkellä näyttää tältä: 
![Screenshot 2023-05-23 at 14 30 32](https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot/assets/18038608/dafcf474-285f-4d98-be4c-2d8e01482400)

Reply:
Hmm, joo. Tämä on vähän ongelmallinen flow nykyisellä toteutuksella. Jos peruu maksun niin voitaisiin kai periaatteessa muuttaa enrollment johonkin cancel-tilaan. Tällöin teoriassa pieni mahdollisuus, että menetät paikan ennen kuin yrität maksua uudestaan. Mutta vaatinee jonkun verran muutoksia tuo cancel-tilan lisääminen esim. paikkojen määrän laskuihin. Täytyy varmaan tästä vielä keskustella.